decoder/path_reader.go

```diff
@@ -11,7 +11,8 @@ import (
 )
 
 type PathReader interface {
-	Paths(ctx context.Context) []lang.Path
+	Paths(ctx context.Context, languageID string) []lang.Path
+	LanguageIDs() []string
```

Comment:
```suggestion
	LanguageIDs() []string
	Paths(ctx context.Context, languageID string) []lang.Path
```
Nitpick - this just reflects the order in which the methods get called.